.github/workflows/test-changed-firestore.yml

```diff
@@ -29,16 +33,104 @@ jobs:
           yarn
       - name: build
         run: yarn build:changed firestore
-      - name: Run tests if firestore or its dependencies has changed
-        run: yarn test:changed firestore
+      - name: Archive build
+        if: ${{ !cancelled() }}
+        run: |
+          tar -cf build.tar .
+          gzip build.tar
+      - name: Upload build archive
+        if: ${{ !cancelled() }}
+        uses: actions/upload-artifact@v3
+        with:
+          name: build.tar.gz
+          path: build.tar.gz
+          retention-days: ${{ env.artifactRetentionDays }}
+
+  compat-test-chrome:
+    name: Test Firestore Compatible
+
+    runs-on: ubuntu-latest
+
+    needs: build
+    steps:
+      - name: Checkout Repo
```

Comment:
Do you actually need to checkout the repo? Wouldn't it get clobbered by build.tar.gz when it's untarred?

Reply:
Removed.